TESTS/mbed_hal/uart_bare_metal_test/uart_test.py

```diff
@@ -0,0 +1,347 @@
+                'PASSED\0']
+
+def log_info(msg):
+    print('[INFO]: %s') % msg
+
+def log_recv(msg):
+    if(VERBOSE == True):
+        print('[RECV]: %s') % msg
+
+def log_send(msg):
+    if(VERBOSE == True):
+        print('[SEND]: %s') % msg
+
+def log_fail(msg):
+    print('[FAIL]: %s') % msg
+    if(serial_port != None and serial_port.isOpen() == True):
+        serial_port.close()
+    sys.exit()
+
+def log_pass(msg):
+    print('[PASS]: %s') % msg
+    if(serial_port != None and serial_port.isOpen() == True):
+        serial_port.close()
+        
+def create_test_header_file(baudrate):
+    """! Create header file with the specified baud rate.
+    
+    @param baudrate specified transmission baud rate.
+    """
+    file = open('test_config.h','w') 
+    file.write('/* This file has been automatically generated by the python script. */\n')
+    file.write('/* This file contains serial port baud rate. */\n')
+    file.write('#define BAUD_RATE %s\n' % baudrate)
+    file.close()
+
+def run_command(cmd):
+    """! Runs command and prints proc stdout on screen.
+         It is used to build the test.
+         
+    @param cmd command to be executed.
+    @return image path.
+    """
+    build_success = True
+    image_path = str()
+    
+    try:
+        p = Popen(cmd, stdout=PIPE, stderr=STDOUT)
+    except OSError as e:
+        log_fail('Command can not be executed.')
+        sys.exit()
+    
+    output = str()
+    
+    for line in iter(p.stdout.readline, b''):
+        if('ERROR' in line):
+            build_success = False
+        if('Image:' in line):
+            image_path = line[7:]
+        output += line
+        # When dumping output to file both \r and \n will be a new line
+        # To avoid this "extra new-line" we only use \n at the end
+        sys.stdout.write(line.rstrip() + '\n')
+        sys.stdout.flush()
+
+    # Check process status.
+    returncode = p.wait()
+    if returncode < 0:
+        log_fail('Error while executing command.')
+        sys.exit()
+    
+    # Check if errors have been detected while building.
+    if(not build_success):
+        log_fail('Error detected while building the test.')
+        sys.exit() 
+    
+    return image_path.strip()
+    
+def reset_mbed_board(port, delay_break, delay_reset):
+    """Perform reset of the mbed device by sending
+       a break signal to the device.
+       
+    @param port serial port id.
+    @param delay_break break signal time [s].
+    @param delay_reset delay time after reset [s].
+    """
+    log_info('Reset the mbed board...')
+    
+    try:
+        if(port.upper().startswith('COM') and int(port[3:]) > 8):
+            port = '\\\\.\\' + port
+    except ValueError:
+        # for like COMnotanumber
+        pass
+    
+    hComPort = win32.CreateFile(port, win32.GENERIC_READ | win32.GENERIC_WRITE, 0, None, win32.OPEN_EXISTING, win32.FILE_ATTRIBUTE_NORMAL | win32.FILE_FLAG_OVERLAPPED, 0)
```

Comment:
The assumption of using a Windows machine should be documented.

Reply:
Note has been added to the script description.

Reply:
You should just not assume that you're on a windows machine! We have linux machines in CI.

Reply:
I agree, but currently I don't have linux machine.